hermes/bin/src/logger.rs

```diff
@@ -0,0 +1,103 @@
+//! Setup for logging for the service.
+
+use std::fmt;
+
+use clap::ValueEnum;
+use tracing::{level_filters::LevelFilter, subscriber::SetGlobalDefaultError};
+use tracing_subscriber::{
+    fmt::{format::FmtSpan, time},
+    FmtSubscriber,
+};
+
+use crate::runtime_extensions::bindings::hermes::logging::api::Level;
+
+/// All valid logging levels.
+#[derive(ValueEnum, Clone, Copy)]
+pub(crate) enum LogLevel {
+    /// Errors
+    Error,
+    /// Warnings
+    Warn,
+    /// Informational Messages
+    Info,
+    /// Debug messages
+    Debug,
+    /// Tracing
+    Trace,
+}
+
+impl fmt::Display for LogLevel {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            LogLevel::Error => write!(f, "Error"),
+            LogLevel::Warn => write!(f, "Warn"),
+            LogLevel::Info => write!(f, "Info"),
+            LogLevel::Debug => write!(f, "Debug"),
+            LogLevel::Trace => write!(f, "Trace"),
+        }
+    }
+}
```

Comment:
You can use [`derive_more`](https://docs.rs/derive_more/latest/derive_more/#) for Display. It's far more maintainable.